src/Analyzers/Core/Analyzers/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -50,91 +66,74 @@ public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
 
         protected abstract bool ShouldAnalyze(Compilation compilation);
 
+        protected abstract TSyntaxKind IfStatementSyntaxKind { get; }
         protected abstract ISyntaxFacts GetSyntaxFacts();
         protected abstract bool IsInExpressionTree(SemanticModel semanticModel, SyntaxNode node, INamedTypeSymbol? expressionTypeOpt, CancellationToken cancellationToken);
 
         protected abstract bool TryAnalyzePatternCondition(
-            ISyntaxFacts syntaxFacts, SyntaxNode conditionNode,
-            [NotNullWhen(true)] out SyntaxNode? conditionPartToCheck, out bool isEquals);
+            ISyntaxFacts syntaxFacts, TExpressionSyntax conditionNode,
+            [NotNullWhen(true)] out TExpressionSyntax? conditionPartToCheck, out bool isEquals);
```

Comment:
a lot more members become more strongly typed to help avoid issues or catch them higher up.